evals/safety_evaluation.py

```diff
@@ -0,0 +1,148 @@
+import argparse
+import asyncio
+import logging
+import os
+import pathlib
+from typing import Any, Dict, List, Optional
+
+import requests
+from azure.ai.evaluation import ContentSafetyEvaluator
+from azure.ai.evaluation.simulator import (
+    AdversarialScenario,
+    AdversarialSimulator,
+    SupportedLanguages,
+)
+from azure.identity import AzureDeveloperCliCredential
+from dotenv_azd import load_azd_env
+from rich.logging import RichHandler
+from rich.progress import track
+
+logger = logging.getLogger("ragapp")
+
+root_dir = pathlib.Path(__file__).parent
+
+
+def get_azure_credential():
+    AZURE_TENANT_ID = os.getenv("AZURE_TENANT_ID")
+    if AZURE_TENANT_ID:
+        logger.info("Setting up Azure credential using AzureDeveloperCliCredential with tenant_id %s", AZURE_TENANT_ID)
+        azure_credential = AzureDeveloperCliCredential(tenant_id=AZURE_TENANT_ID, process_timeout=60)
+    else:
+        logger.info("Setting up Azure credential using AzureDeveloperCliCredential for home tenant")
+        azure_credential = AzureDeveloperCliCredential(process_timeout=60)
+    return azure_credential
+
+
+async def callback(
+    messages: List[Dict],
+    stream: bool = False,
+    session_state: Any = None,
+    context: Optional[Dict[str, Any]] = None,
+    target_url: str = "http://localhost:50505/chat",
+):
+    messages_list = messages["messages"]
+    latest_message = messages_list[-1]
```

Comment:
chance this crashes on empty messages?

Reply:
The messages here come from the simulator itself, so they should never be empty